cabal-install/Distribution/Client/SetupWrapper.hs

```diff
@@ -604,16 +605,22 @@ externalSetupMethod verbosity options pkg bt mkargs = do
           -- With 'useDependenciesExclusive' we enforce the deps specified,
           -- so only the given ones can be used. Otherwise we allow the use
           -- of packages in the ambient environment, and add on a dep on the
-          -- Cabal library.
+          -- Cabal library (unless 'useDependencies' already contains one).
           --
           -- With 'useVersionMacros' we use a version CPP macros .h file.
           --
           -- Both of these options should be enabled for packages that have
           -- opted-in and declared a custom-settup stanza.
           --
```

Comment:
Ugh, this is annoying. Under what circumstances do we have `not useDependenciesExclusive` but we DO have a Cabal? (I guess something related to the default setup deps?)